packages/core/src/Stores/client-store.js

```diff
@@ -1252,6 +1253,32 @@ export default class ClientStore extends BaseStore {
         }
     }
 
+    @action.bound
+    fetchConnectedApps() {
+        return new Promise(async (resolve, reject) => {
+            const response = await WS.send({ oauth_apps: 1 });
+            if (!response.error) {
+                resolve(response.oauth_apps);
+            } else {
+                reject(response.error);
+            }
+        });
+    }
+    @action.bound
+    setConnectedApps(apps_list = []) {
+        this.connected_apps = apps_list;
```

Comment:
why don't we just do this in the same function fetchConnectedApps?

Reply:
I just want to write it more functional but, sure :+1: